src/output.rs

```diff
@@ -59,7 +59,20 @@ impl Output {
         for filter in &self.filters {
             let f = try!(context.get_filter(&filter.name)
                 .ok_or(Error::Render(format!("Filter {} not implemented", &filter.name))));
-            entry = try!(f(&entry, &filter.arguments));
+
+            let mut arguments = Vec::new();
+            for arg in filter.arguments.iter() {
+                match arg {
+                    &VarOrVal::Var(ref x) => {
+                        let val = context.get_val(&*x.name()).cloned().unwrap_or(Value::Str("".to_owned()));
```

Comment:
instead of defaulting to a `Value::Str` you should throw a render error saying that the variable is not defined